### PR TITLE
Update tN Progress on GL Selection

### PR DESCRIPTION
**Pull request by RoyalSix**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- This PR adds functionality to update the tN tool progress after selecting a gl

#### Please include detailed Test instructions for your pull request:
- Follow test instructions on #6438 and ensure the progress get updated
- Ensure normal tool functionality is working, especially the loading process (Local/Online)

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6469)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (99281ce)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-linux-x64-2.0.0-99281ce.deb) is ready for download.
[macOS build 2.0.0 (99281ce)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-macos-x64-2.0.0-99281ce.dmg) is ready for download.
[Windows x64 build 2.0.0 (99281ce)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x64-2.0.0-99281ce.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (99281ce)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x32-2.0.0-99281ce.setup.exe) is ready for download.

File: src/js/actions/ProjectDetailsActions.js [hunk 1]
```diff
@@ -131,15 +131,22 @@ export function setProjectToolGL(toolName, selectedGL) {
...
-
     dispatch({
       type: consts.SET_GL_FOR_TOOL,
       toolName,
       selectedGL,
     });
+
+    if (toolName === TRANSLATION_NOTES) { // checks on tN are based on GL, but tW is based on OrigLang so don't need to be updated on GL change
+      dispatch(ResourcesHelpers.updateGroupIndexForGl(toolName, selectedGL));
+      dispatch(prepareToolForLoading(toolName)).then(() => {
```
Comment (mannycolon) [inline src/js/actions/ProjectDetailsActions.js]: Would using async/await make this code easier to read?

File: src/js/actions/ProjectDetailsActions.js [hunk 2]
```diff
@@ -149,33 +156,18 @@ export function setProjectToolGL(toolName, selectedGL) {
...
     try {
-      if (toolName === WORD_ALIGNMENT) {
-        const toolApi = getToolsByKey(getState());
-        const currentToolApi = toolApi[toolName].api;
-        //TODO: add progress fetch code to checking-tool-wrapper
-        progress = currentToolApi.trigger('getProgress');
-      } else {
-        progress = ProjectDetailsHelpers.getToolProgress(pathToCheckDataFiles, toolName, toolsCategories[toolName], bookId);
-      }
+      const toolApi = getToolsByKey(getState());
```
Comment (mannycolon) [inline src/js/actions/ProjectDetailsActions.js:167]: I assume the getProgress method was added to checking-tool-wrapper?

File: src/js/actions/GroupsDataActions.js [hunk 3]
```diff
@@ -53,11 +53,11 @@ export const findGroupDataItem = (contextId, groupData) => {
  * @description verifies that the data in the checkdata folder is reflected in the menu.
  * @return {object} action object.
  */
-export function verifyGroupDataMatchesWithFs() {
+export function verifyGroupDataMatchesWithFs(toolName) {
```
Comment (mannycolon) [inline src/js/actions/GroupsDataActions.js:69]: would it be a good idea to add a default value to toolName? That way we avoid the undefined value.

verifyGroupDataMatchesWithFs(toolName = null)

File: src/js/actions/ContextIdActions.js [hunk 4]
```diff
@@ -159,12 +159,12 @@ export const changeToPreviousContextId = () => ((dispatch, getState) => {
  * @description loads the latest contextId file from the file system.
  * @return {object} Dispatches an action that loads the contextId with data.
  */
-export function loadCurrentContextId() {
+export function loadCurrentContextId(toolName) {
```
Comment (mannycolon) [inline src/js/actions/ContextIdActions.js:162]: ibid

File: __tests__/ProjectDetailsActions.test.js [hunk 5]
```diff
@@ -461,6 +471,7 @@ describe('ProjectDetailsActions.updateCategorySelection', () => {
       progress: 0.25,
     }];
     const store = mockStore(initialState);
+    console.log('store', store.getState());
```
Comment (mannycolon) [inline __tests__/ProjectDetailsActions.test.js]: Do we still need to log the store here?

**Review by mannycolon — CHANGES_REQUESTED** [review]

**Comment by mannycolon** [comment]
@RoyalSix Please connect this PR to the issue it belongs to

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (dce19b1)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-linux-x64-2.0.0-dce19b1.deb) is ready for download.
[macOS build 2.0.0 (dce19b1)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-macos-x64-2.0.0-dce19b1.dmg) is ready for download.
[Windows x64 build 2.0.0 (dce19b1)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x64-2.0.0-dce19b1.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (dce19b1)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x32-2.0.0-dce19b1.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (dce19b1)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-linux-x64-2.0.0-dce19b1.deb) is ready for download.
[macOS build 2.0.0 (dce19b1)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-macos-x64-2.0.0-dce19b1.dmg) is ready for download.
[Windows x64 build 2.0.0 (dce19b1)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x64-2.0.0-dce19b1.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (dce19b1)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x32-2.0.0-dce19b1.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (a9ec14f)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-linux-x64-2.0.0-a9ec14f.deb) is ready for download.
[macOS build 2.0.0 (a9ec14f)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-macos-x64-2.0.0-a9ec14f.dmg) is ready for download.
[Windows x64 build 2.0.0 (a9ec14f)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x64-2.0.0-a9ec14f.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (a9ec14f)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x32-2.0.0-a9ec14f.setup.exe) is ready for download.

File: src/js/actions/ContextIdActions.js [hunk 6]
```diff
@@ -159,12 +159,12 @@ export const changeToPreviousContextId = () => ((dispatch, getState) => {
  * @description loads the latest contextId file from the file system.
  * @return {object} Dispatches an action that loads the contextId with data.
  */
-export function loadCurrentContextId() {
+export function loadCurrentContextId(toolName = '') {
```
Comment (mannycolon) [inline src/js/actions/ContextIdActions.js]: Should toolName default to null?

Comment (RoyalSix) [reply src/js/actions/ContextIdActions.js]: Yea, either works. I updated.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (7eed8a4)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-linux-x64-2.0.0-7eed8a4.deb) is ready for download.
[macOS build 2.0.0 (7eed8a4)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-macos-x64-2.0.0-7eed8a4.dmg) is ready for download.
[Windows x64 build 2.0.0 (7eed8a4)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x64-2.0.0-7eed8a4.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (7eed8a4)](https://tc-builds.door43.org/build/feature-jay-6438/translationCore-win-x32-2.0.0-7eed8a4.setup.exe) is ready for download.

**Review by RoyalSix — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6469)* status: 0 of 8 files reviewed, 6 unresolved discussions (waiting on @mannycolon)

___
*[__tests__/ProjectDetailsActions.test.js, line 474 at r1](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6469#-Lpo13XEBRyTZOG88cnd-r1-474:-Lq7ReWH9hr0mJSRp2C_:b-896fix) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/99281cef36a98724ee98b4ffbe948b4d9b45d4c1/__tests__/ProjectDetailsActions.test.js#L474)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

Do we still need to log the store here?
</blockquote></details>

Done.
___
*[src/js/actions/ContextIdActions.js, line 162 at r1](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6469#-Lpo13XEBRyTZOG88cnf-r1-162:-Lq7Rew98wwqP7p96vgj:b-896fix) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/99281cef36a98724ee98b4ffbe948b4d9b45d4c1/src/js/actions/ContextIdActions.js#L162)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

ibid
</blockquote></details>

Done.
___
*[src/js/actions/GroupsDataActions.js, line 56 at r1](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6469#-Lpo13XEBRyTZOG88cng-r1-56:-Lq7RfV51uMLeE9-3ZkH:b-896fix) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/99281cef36a98724ee98b4ffbe948b4d9b45d4c1/src/js/actions/GroupsDataActions.js#L56)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

would it be a good idea to add a default value to toolName? That way we avoid the undefined value.

verifyGroupDataMatchesWithFs(toolName = null) 
</blockquote></details>

Done.
___
*[src/js/actions/ProjectDetailsActions.js, line 142 at r1](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6469#-Lpo13XEBRyTZOG88cnh-r1-142:-Lq7RgDjEeRRjL5Qruou:b-896fix) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/99281cef36a98724ee98b4ffbe948b4d9b45d4c1/src/js/actions/ProjectDetailsActions.js#L142)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

Would using async/await make this code easier to read? 
</blockquote></details>

Done.
___
*[src/js/actions/ProjectDetailsActions.js, line 168 at r1](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6469#-Lpo13XEBRyTZOG88cnh-r1-168:-Lq7Rgkn2SYTanFwzwjX:bn2xdt6) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/99281cef36a98724ee98b4ffbe948b4d9b45d4c1/src/js/actions/ProjectDetailsActions.js#L168)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

I assume the getProgress method was added to checking-tool-wrapper? 
</blockquote></details>

Yes, added default value if the method doesn't exist (it will return undefined) https://github.com/translationCoreApps/tc-tool/blob/develop/src/api/Lifecycle.js#L88

**Pull request merged**